classifiers/text_analysis/emotionality_detection/code_snippet_common.md

```diff
@@ -1,26 +1,38 @@
 ```python
-from LeXmo import LeXmo
 
-def emotionality_detection(text:str) -> str:
-    """
-    @param text: text to check
-    @return: either 'anger', 'fear', 'anticipation', 'trust', 'surprise', 'sadness', 'joy' or 'disgust' depending on the score
-    """
-    emo = LeXmo.LeXmo(text)
-    del emo["text"]
-    del emo["positive"]
-    del emo["negative"]
-    emo = max(emo, key=emo.get)
-    return emo
-    
+import requests
+import spacy
+
+def emotionality_detection(text, api_key):
+      headers = {"Authorization": f"Bearer {api_key}"}
+      data = {"inputs": text}
+      try: 
+            response = requests.post("https://api-inference.huggingface.co/models/j-hartmann/emotion-english-distilroberta-base", headers=headers, json=data)
+            response_json = response.json()
+            ner_positions = []
+
+            nlp = spacy.load("en_core_web_sm")
+            doc = nlp(text)
+
+            for item in response_json:
+                  start = item["start"]
+                  end = item["end"]
+                  span = doc.char_span(start, end, alignment_mode="expand")
+                  ner_positions.append((item["entity_group"], span.start, span.end))
+            return ner_positions
+      except Exception as e: 
+            return f"That didn't work. Did you provide a valid API key? Go error: {e} and message: {response_json}"
+
 # ↑ necessary bricks function 
 # -----------------------------------------------------------------------------------------
 # ↓ example implementation 
 
-def example_integration():
-    texts = ["I am scared to continue.", "Oh my goodness it was the best evening ever, hype!"]
-    for text in texts:
-        print(f"\"{text}\" has emotion: {emotionality_detection(text)}")
+def emotionality_detection():
+      hf_api_key = "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"
```

Comment:
@SvenjaKern Please watch out to **never** include API keys in your git commits. Please deactivate this API key and generate a new one. This is a public repository, which are often scanned for keys like this which can then be abused.